MailCore/Utils/MessageBodyUtils.swift

```diff
@@ -118,6 +145,85 @@ public enum MessageBodyUtils {
 
+    private static func createPrintHeader(message: Message) throws -> Element {
+        let rootHeaderDiv = try Element(Tag("div"), "").attr("id", "printHeader")
+        let firstSeparator = try Element(Tag("hr"), "")
+            .attr("size", "1")
+            .attr("color", "black")
+            .attr("width", "100%")
+            .attr("style", "margin-bottom: 10px")
+        let secondSeparator = try Element(Tag("hr"), "")
+            .attr("size", "1")
+            .attr("color", "LightGray")
+            .attr("width", "100%")
+            .attr("style", "margin-top: 10px")
+            .attr("style", "margin-bottom: 10px")
+
+        let b64image = MailResourcesAsset.logoText.image.pngData()?.base64EncodedString()
+
+        let iconElement = try Element(Tag("img"), "")
+            .attr("src", "data:image/png;base64, \(b64image ?? "")")
+            .attr("style", "margin-bottom: 10px; width: 150px;")
+
+        try rootHeaderDiv.insertChildren(0, [iconElement, firstSeparator])
+
+        if let subject = message.subject {
+            let subjectElement = try Element(Tag("b"), "").appendText(subject)
+            try rootHeaderDiv.insertChildren(2, [subjectElement])
+        }
+
+        try rootHeaderDiv.insertChildren(3, [secondSeparator])
+
+        var messageDetailsDiv = try Element(Tag("div"), "").attr("style", "margin-bottom: 40px")
+        messageDetailsDiv = try insertPrintRecipientField(
+            to: messageDetailsDiv,
+            prefix: MailResourcesStrings.Localizable.ccTitle,
+            recipients: message.cc.toArray()
+        )
+        messageDetailsDiv = try insertPrintRecipientField(
+            to: messageDetailsDiv,
+            prefix: MailResourcesStrings.Localizable.toTitle,
+            recipients: message.to.toArray()
+        )
+        messageDetailsDiv = try insertPrintRecipientField(
+            to: messageDetailsDiv,
+            prefix: MailResourcesStrings.Localizable.fromTitle,
+            recipients: message.from.toArray()
+        )
+        messageDetailsDiv = try insertPrintDateField(
+            to: messageDetailsDiv,
+            prefix: MailResourcesStrings.Localizable.dateTitle,
+            date: message.date.formatted(date: .long, time: .shortened)
+        )
+
+        try rootHeaderDiv.insertChildren(4, [messageDetailsDiv])
+        return rootHeaderDiv
+    }
+
+    private static func insertPrintRecipientField(to element: Element, prefix: String,
+                                                  recipients: [Recipient]) throws -> Element {
+        guard !recipients.isEmpty else { return element }
+
+        let recipientsField = Element(Tag("div"), "")
+        let fieldName = try Element(Tag("b"), "").appendText(prefix)
+        let recipientsName = recipients.map { $0.htmlDescription }.joined(separator: ", ")
+        let fieldValue = try Element(Tag("text"), "").appendText(recipientsName)
+
+        try recipientsField.insertChildren(0, [fieldName, fieldValue])
```

Comment:
Wouldn't it be easier to write HTML in a string, and then use SwiftSoup to parse the code fragment?